lib/http/response.js

```diff
@@ -53,14 +53,11 @@ class HttpResponse extends EventEmitter {
       if (this.request.isAborted) {
         return;
       }
-
-      this.emit('success', this.responseData, this.res, this.redirected);
-
-      // if (this.res.statusCode.toString().indexOf('2') === 0 || this.redirected) {
-      //
-      // } else {
-      //   this.emit('error', this.responseData, this.res, screenshotContent);
-      // }
+      if (this.res.statusCode.toString().indexOf('2') === 0 || this.redirected) {
```

Comment:
Not sure about these changes to the http module. What is the issue you're solving here?

Reply:
the error event was never emitted so inorder to verify this testcase I made this change.
[TEST](https://github.com/gravityvi/nightwatch/blob/76e555a666ae7ddf40e2167506338789320f31d8/test/src/index/testRequest.js#L189-L189)

Reply:
I think this change was made in order to be closer to how the selenium http client works. I think we need to keep it like this, because the purpose of the error event is that the request should be retried, and (if I remember correctly, in the selenium http client) it should only be retried if it's a client error. You can check what they did here: https://github.com/SeleniumHQ/selenium/blob/trunk/javascript/node/selenium-webdriver/http/index.js#L159

Reply:
There are multiple tests that depend on this change. should we remove those?
for example: https://github.com/nightwatchjs/nightwatch/blob/f7dc9b4ffde5aafe0e9df5089971a668d2c2cb29/test/src/index/testRequest.js#L189